Development-Guide/Compile-and-Flash-STMBoard-Guide/README.adoc

```diff
@@ -54,15 +56,56 @@ For example, EWARM is for the IAR Embedded Workbench IDE and CubeMX2Makefile wil
 
 . Go to the directory that the Makefile was made then compile using `make`.
 
-. Navigate to the build directory and there should be a `<project>.bin` file. Where `<project>` is the project name. 
+. Navigate to the build directory and there should be a `<project>.bin` & a `<project>.elf` file. Where `<project>` is the project name. 
 
 == Flashing
-Flashing is how the `.bin` that was made is put onto the board. This is how the processor on the board is able to carry out the instructions that were generated from the code.
+Flashing is how the `.bin` or `.elf` that was made is put onto the board. 
+This is how the processor on the board is able to carry out the instructions that were generated from the code.
+
+There are two methods of flashing the boards - using the ST-Link Programmer, or using the Black Magic Probe
+
+=== ST-Link Programmer
 
-To flash the board use `st-flash write /path/to/project/build/<project>.bin 0x8000000`.
+Connect the programmer to the board.
+
+Flash the board using `st-flash write /path/to/project/build/<project>.bin 0x8000000`.
 
 If a board isn't connected, and it ran successfully, the output should be: 
 ----
 st-flash 1.3.0-dev
 DATE/TIME WARN /opt/stlink/src/usb.c: Couldn't find any ST-Link/V2 devices
 ----
+
+=== Black Magic Probe
+
+The Black Magic Probe is compatable both the 10-Pin & 20-Pin JTag headers. 
+It also allows for debugging using GDB.
+
+To load the `.elf` file onto the board, enter the `build/` directory.
+
+. Plug in the Probe to the board and into your computer.
+Determine where the probe is located on your computer, usually `/dev/ttyACM<X>`, where `<X>` is a number.
+Typically, there are two devices listed there. 
+Make note of the lowest value number.
+
+. Use the command  sudo arm-none-eabi-gdb `<project>.elf`
```

Comment:
Should this be `sudo arm-none-eabi-gdb <project>.elf` not sudo arm-none-eabi-gdb `<project>.elf`?